nodejs/eks/cluster.ts

```diff
@@ -17,13 +17,54 @@ import * as k8s from "@pulumi/kubernetes";
 import * as pulumi from "@pulumi/pulumi";
 import * as crypto from "crypto";
 import * as fs from "fs";
+import * as jsyaml from "js-yaml";
 import * as path from "path";
 import which = require("which");
 
 import { ServiceRole } from "./servicerole";
 import { createStorageClass, EBSVolumeType, StorageClass } from "./storageclass";
 import transform from "./transform";
 
+/**
+ * RoleMapping describes a mapping from an AWS IAM role to a Kubernetes user and groups.
+ */
+export interface RoleMapping {
+    /**
+     * The ARN of the IAM role to add.
+     */
+    roleArn: pulumi.Input<aws.ARN>;
+
+    /**
+     * The user name within Kubernetes to map to the IAM role. By default, the user name is the ARN of the IAM role.
+     */
+    username: pulumi.Input<string>;
+
+    /**
+     * A list of groups within Kubernetes to which the role is mapped.
+     */
+    groups: pulumi.Input<pulumi.Input<string>[]>;
```

Comment:
Actually, one thing, this type looks like it's on purpose, but it's not clear to me why not just make it `pulumi.Input<string[]>`

Reply:
It's a little clearer if you write `X[]` as `Array<X>` - `pulumi.Input<Array<pulumi.Input<string>>`.  That is, the array itself can be an Output, or the elements of the array can be Outputs.

Reply:
@lukehoban mmm, yes, that makes sense and seems obvious in retrospect that you'd want to do this.